sentry-core/src/main/java/io/sentry/core/cache/SessionCache.java

```diff
@@ -135,7 +135,11 @@ public void store(final @NotNull SentryEnvelope envelope, final @Nullable Object
             }
 
             session.end(timestamp);
-            final SentryEnvelope fromSession = SentryEnvelope.fromSession(serializer, session);
+            // if the App. has been upgraded and there's a new version of the SDK running, SdkInfo
```

Comment:
L: What about we store the sdk-info with the session data?

Reply:
I'll create a new task for that and not totally related to this one as this is a very corner case.
I think we can tackle that along with the session refactoring+store endpoint task.